### PR TITLE
support Pardiso in Eigen

### PR DESCRIPTION
**Pull request by norihiro-w**
Credit goes to @chleh, because it's based on his branch (see #1175) and enables to use Pardiso solver with Eigen interface. To use it, one needs to set CMake as `-DOGS_USE_EIGEN=ON -DOGS_USE_MKL=ON -DMKL_DIR=(a path to MKL root directory)`. In a config file, you can set `PardisoLU` as a solver type.
 
@chleh Let me know if I miss something from your branch.

### CODE REVIEW

File: scripts/cmake/cmake/FindMKL.cmake [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Find Intel Math Karnel Library (MKL)
```
Comment (endJunction) [inline scripts/cmake/cmake/FindMKL.cmake]: Either  a source of this file, or our copyright message is missing. (Though the other cmakes don't have it too, but let's start here.)

**Comment by chleh** [comment]
:+1: Looks good. Thanks for the credits, but actually it looks like you completely did it on your own (although I don't remember my code anymore).

When testing Eigen+MKL I think there were severe issues if I used Eigen's BiCGStab multithreaded with OpenMP (i.e. it crashed). Have you tried it? Did you see crashes?

**Comment by norihiro-w** [comment]
@chleh No, I haven't tried the multi-threaded BiCGStab.

**Comment by norihiro-w** [comment]
@bilke Jenkins job "default" is pending for a long time. what's going on here?

**Comment by bilke** [comment]
The default job are the Linux-PR-Dynamic jobs on the [UFZ internal Jenkins](https://svn.ufz.de:8443/) which is currently down. I do not know why. The jobs run on envinf1 which was done the last days too.

**Comment by norihiro-w** [comment]
@bilke i see. then I wait until the internal Jenkins is back

**Pull request merged**